backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/Controllers/RoleController.cs

```diff
@@ -21,179 +18,163 @@ public class RoleController : Controller
             _logger = logger;
             _roleService = roleService;
-            _serializerOptions.Converters.Add(new JsonStringEnumConverter());
-            _featureFlags = featureFlags.Value;
         }
 
         /// <summary>
-        ///     Search through all roles and return matches
+        /// Gets role connections for the given parties.
         /// </summary>
-        /// <returns>All search results, sorted into areas</returns>
+        /// <param name="party">The party performing the lookup.</param>
+        /// <param name="from">Optional right owner to filter on.</param>
+        /// <param name="to">Optional right holder to filter on.</param>
         [HttpGet]
         [Authorize]
-        [Route("search")]
-        public async Task<ActionResult<List<RoleAreaFE>>> Search([FromQuery] string searchString)
+        [Route("connections")]
+        public async Task<ActionResult<List<RolePermission>>> GetConnections([FromQuery] Guid party, [FromQuery] Guid? from, [FromQuery] Guid? to)
         {
             if (!ModelState.IsValid)
             {
                 return BadRequest(ModelState);
             }
 
-            var languageCode = LanguageHelper.GetSelectedLanguageCookieValueBackendStandard(_httpContextAccessor.HttpContext);
-            try
-            {
-                return await _roleService.GetSearch(languageCode, searchString);
-            }
-            catch (HttpStatusException ex)
+            if (!from.HasValue && !to.HasValue)
             {
-                if (ex.StatusCode == HttpStatusCode.NoContent)
-                {
-                    return NoContent();
-                }
-
-                string responseContent = ex.Message;
-                return new ObjectResult(ProblemDetailsFactory.CreateProblemDetails(HttpContext, (int?)ex.StatusCode, "Unexpected HttpStatus response", detail: responseContent));
+                return BadRequest("Either 'from' or 'to' query parameter must be provided.");
             }
-        }
 
-        /// <summary>
-        ///     Gets meta data for a given role
-        /// </summary>
-        /// <returns>The role meta data for </returns>
-        [HttpGet]
-        [Authorize]
-        [Route("{id}")]
-        public async Task<ActionResult<Core.Models.Common.Role>> GetRoleMetaById([FromRoute] Guid id)
-        {
-            if (!ModelState.IsValid)
-            {
-                return BadRequest(ModelState);
-            }
-
-            var languageCode = LanguageHelper.GetSelectedLanguageCookieValueBackendStandard(_httpContextAccessor.HttpContext);
             try
             {
-                return await _roleService.GetRoleMetaById(languageCode, id);
+                string languageCode = LanguageHelper.GetSelectedLanguageCookieValueBackendStandard(_httpContextAccessor.HttpContext);
+                List<RolePermission> connections = await _roleService.GetConnections(party, from, to, languageCode);
```

Comment:
```suggestion
                List<RolePermission> connections = await _roleService.GetPermissions(party, from, to, languageCode);
```
The same goes for the Service